fearless/Modules/Staking/Analytics/Rewards/AnalyticsRewardsInteractor.swift

```diff
@@ -0,0 +1,76 @@
+import UIKit
+import RobinHood
+
+final class AnalyticsRewardsInteractor {
+    weak var presenter: AnalyticsRewardsInteractorOutputProtocol!
+
+    let singleValueProviderFactory: SingleValueProviderFactoryProtocol
+
+    private let analyticsService: AnalyticsService?
+    private let assetId: WalletAssetId
+    private let substrateProviderFactory: SubstrateDataProviderFactoryProtocol
+    private let selectedAccountAddress: AccountAddress
+    private var priceProvider: AnySingleValueProvider<PriceData>?
+    private var stashControllerProvider: StreamableProvider<StashItem>?
+
+    init(
+        singleValueProviderFactory: SingleValueProviderFactoryProtocol,
+        analyticsService: AnalyticsService?,
+        assetId: WalletAssetId,
+        substrateProviderFactory: SubstrateDataProviderFactoryProtocol,
+        selectedAccountAddress: AccountAddress
+    ) {
+        self.singleValueProviderFactory = singleValueProviderFactory
+        self.analyticsService = analyticsService
+        self.assetId = assetId
+        self.substrateProviderFactory = substrateProviderFactory
+        self.selectedAccountAddress = selectedAccountAddress
+    }
+
+    private func fetchAnalyticsRewards() {
+        // TODO: delete stub data
+        let timestamp = Int64(Date().timeIntervalSince1970)
+        let stubData = (1 ..< 100).map {
+            SubqueryRewardItemData(amount: $0.description, isReward: true, timestamp: timestamp - $0 * 10000)
+        }
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1) { [weak self] in
+            self?.presenter?.didReceieve(rewardItemData: .success(stubData))
+        }
+    }
+
+    private func subscribeToStashControllerProvider() {
```

Comment:
I think we have `subscribeToStashItemProvider` in `SubstrateProviderSubscriber`. Can we use it?